Sources/Vehicle/Templates/CarPlayPaginatedListTemplate.swift

```diff
@@ -0,0 +1,98 @@
+import CarPlay
+import Foundation
+import Shared
+
+final class CarPlayPaginatedListTemplate {
+    enum GridPage {
+        case next
+        case previous
+    }
+
+    private var items: [CPListItem]
+    private var currentPage: Int
+    private let title: String
+
+    private let itemsPerPage: Int = CPListTemplate.maximumItemCount
+    private(set) var template: CPListTemplate
+
+    init(title: String, items: [CPListItem]) {
+        self.title = title
+        self.items = items
+        self.currentPage = 0
+        self.template = CPListTemplate(title: title, sections: [])
+    }
+
+    func updateItems(items: [CPListItem], refreshUI: Bool = false) {
```

Comment:
what is the purpose of 'refreshUI' as an option here?

Reply:
Refresh UI would literally update all items from the list, which causes a glitch that scrolls the user to the top and move the cursor (for non-touchscreen carplays), so this is only used when you are setting the list for the first time, I will add a comment to explain that